testing/unittests/esdltests.cpp

```diff
@@ -1844,6 +1870,220 @@ constexpr const char * result = R"!!(<soap:Envelope xmlns:soap="http://schemas.x
+              <es:parameter name="ddd" value="ddd"/>
+              <es:parameter name="u1" value="u1"/>
+              <es:parameter name="u2" value="u2"/>
+              <es:parameter name="dt" value="dt"/>
+              <es:sql>INSERT INTO tbl1 values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);</es:sql>
+            </es:mysql>
+            <es:mysql server="$server" user="$user" password="$password" database="$database" section="$section" name="insert_inline">
+              <es:sql>INSERT INTO tbl1 values ('inline', 65, 1, 1, 1.2, 3.4, 'aa55aa55', 1234567.89, 'Straße', 'Straße', '2019-02-01 23:59:59');</es:sql>
+            </es:mysql>
+            <es:mysql server="$server" user="$user" password="$password" database="$database" section="$section" name="select_all">
+              <es:sql>SELECT * FROM tbl1</es:sql>
+            </es:mysql>
+        </es:CustomRequestTransform>
+        )!!";
+
+        constexpr const char *config1 = R"!!(<config>
+          <Transform>
+            <Param name='testcase' value="new features"/>
+          </Transform>
+        </config>)!!";
+
+        static constexpr const char * data = R"!!(<?xml version="1.0" encoding="UTF-8"?>
+         <root>
+            <insert>
+             <common_value>178</common_value>
+             <common_r8>1.2</common_r8>
+             <Row>
+               <name>selected1</name>
+               <bval>65</bval>
+               <boolval>1</boolval>
+               <r4>3.4</r4>
+               <d>aa55aa55</d>
+               <ddd>1234567.89</ddd>
+               <u1>Straße1</u1>
+               <u2>ᚠᛇᚻ᛫ᛒᛦᚦ᛫ᚠᚱᚩᚠᚢᚱ᛫ᚠᛁᚱᚪ᛫ᚷᛖᚻᚹᛦᛚᚳᚢᛗ</u2>
+               <dt>2019-02-01 12:59:59</dt>
+             </Row>
+             <Row>
+               <name>selected2</name>
+               <bval>65</bval>
+               <boolval>1</boolval>
+               <r4>4.5</r4>
+               <d>bb66bb66</d>
+               <ddd>1234567.89</ddd>
+               <u1>Straße3</u1>
+               <u2>Straße4</u2>
+               <dt>2019-02-01 13:59:59</dt>
+             </Row>
+             <Row>
+               <name>selected3</name>
+               <bval>65</bval>
+               <boolval>1</boolval>
+               <r4>5.6</r4>
+               <d>cc77cc77</d>
+               <ddd>1234567.89</ddd>
+               <u1>Straße5</u1>
+               <u2>色は匂へど 散りぬるを</u2>
+               <dt>2019-02-01 14:59:59</dt>
+             </Row>
+            </insert>
+            <read>
+             <name>selected1</name>
+             <name>selected3</name>
+            </read>
+          </root>
+        )!!";
+
+        static constexpr const char * input = R"!!(<?xml version="1.0" encoding="UTF-8"?>
+         <root>
+            <Person>
+               <FullName>
+                  <First>Joe</First>
+               </FullName>
+            </Person>
+          </root>
+        )!!";
+
+        static constexpr const char * script = R"!!(<es:CustomRequestTransform xmlns:es="urn:hpcc:esdl:script" target="Person">
```

Comment:
An offline response to the question of error handling pointed out that errors are recorded in the script context. I've managed to create multiple errors that are not recorded. My offense was that I omitted the apostrophes around several attribute values to distinguish literal text from xpaths. We need a record of errors, including errors that prevent our preferred error handling from creating a record of errors.

Reply:
Database and system errors should be reported in the result.
How we handle coding errors is a different thing and needs thinking about.  It's hard to tell what the intention of the script writer was.  value="x" could mean get the value from the element x.  or it could be a coding mistake and the developer meant to say value="'x'".

Reply:
In my case, an exception was being thrown, caught, and consumed. The exception text, once I saw it, immediately gave me the answer. I was kind of hoping for the exception to at least be added to trace output so I didn't have to debug the process to figure out my mistake.

Reply:
I can def add exceptions to trace.  Longer term we need better ways of detecting script bugs.

Reply:
I think I found why some exceptions weren't in the trace log, and have changed it.  Let me know if there are still things that should be but aren't being trace logged.